src/service/resources-impl.js

```diff
@@ -1201,11 +1212,16 @@ export class Resources {
           this.viewer_.sendMessage('documentHeight',
               dict({'height': measuredContentHeight}), /* cancelUnsent */true);
           this.contentHeight_ = measuredContentHeight;
-          dev().fine(TAG_, 'document height changed: ' + this.contentHeight_);
+          dev().fine(TAG_, 'document height changed: %s', this.contentHeight_);
           this.viewport_.contentHeightChanged();
         }
       });
     }
+
+    this.passCallbacks_.forEach(callback => {
```

Comment:
Runtime has a different set of performance requirements. Please use a for-loop here.

Also when you fix this, don't do `passCallbacks[i]()` because that'll leak `passCallbacks` as the `this` context of the function invocation. Store `fn = passCallbacks[i]`, then invoke `fn()`.